### DISCUSSION TITLE
mac os sequoia 15 - rustdesk no longer opens

### DISCUSSION THREAD

**Discussion by causalityloop** [rustdesk/rustdesk#9412] [General]
### Bug Description

just upgraded to sequoia 15 today and after the upgrade rustdesk no longer opens. If i try to open from the dock, no UI is displayed nor does it show along the clock on the top right (ie service). if i try to open from the CLI (open -n /Applications/RustDesk.app), the result is the same and there is also no output in the terminal

I tried the comment linked [here ](https://github.com/rustdesk/rustdesk/discussions/5631#discussioncomment-6943538)with the same results. Also, tried to uninstall and install nightly with the same result.

### How to Reproduce

upgraded to mac os sequoia 15

### Expected Behavior

service icon showing along the clock (top right) or a UI opens

### Operating system(s) on local side and remote side

osx sequoia 15

### RustDesk Version(s) on local side and remote side

1.30 and 1.31 nightly

### Screenshots

![2024-09-19_19h06_23](https://github.com/user-attachments/assets/7fb55dfb-8fb3-46b1-9368-b59cc9703a88)

**Comment by rustdesk** [comment]
It is ok on my macos 15

**Reply by Aymen-Ben-S** [reply]
How can we enable verbosity to understand where the issue please?

**Comment by 1e0ng** [comment]
The unsigned app not open, but the signed one works.

**Reply by Aymen-Ben-S** [reply]
Where did you find a signed one?

**Comment by Rene-Auberger** [comment]
I have a MacBook with M1 Max chip and the app from https://github.com/rustdesk/rustdesk/releases/download/1.3.1/rustdesk-1.3.1-aarch64.dmg does not open. Trying to open it from the terminal only gives a "killed" response immediately. Where do I find a signed app for M1?

**Comment by Aymen-Ben-S** [comment]
Is there an update in here? How can I enable verbosity when starting in mac?

**Comment by diraneyya** [comment]
> _After a full day of attempting to figure out how to build RustDesk, I seem to have finally succeeded. I basically followed the **GitHub Actions workflows**. Below is the solution I found to the issue of this thread along with the steps I used to build the app, which I would like to share here to ease some of the difficulty other [seem to have been facing](https://github.com/rustdesk/rustdesk/discussions/763)._

After building the unsigned app, it wasn't opening on my Mac OS X 15 as suggested by others here, which led me to this issue. It also won't appear on the list of "unverified" applications in the "Settings" after supposedly-being shutdown by the OS (given this to be the case, which it was!).

**The solution seems to be in checking this single check box in Xcode:**

<div style="text-align: center"><img width="66%" alt="image" src="https://github.com/user-attachments/assets/5207b5ab-f353-48b8-ac99-9ea58615f8c6" /></div>

The checkbox is labeled as `Disable Library Validation`.

After checking that box I was able to build RustDesk on a MacOS X, which worked!

Below I changed the build number to `99999999`:

<img width="66%" alt="image" src="https://github.com/user-attachments/assets/f4944d91-f889-46ba-a58a-6ea99a72f4d9" />


### Further information

For those who might be frustrated with building RustDesk on an Apple Silicone MacOS X 15. I am using `asdf` with the following setup.

#### Ruby & Cocoapods

Replace system `ruby` and install `cocoapods` using `brew`

```bash
brew install cocoapods         # installs newer ruby + cocoapods
brew link --overwrite ruby    # makes the newer ruby accessible when typing `ruby`
```

#### Flutter & Rust Installation

Use `asdf` to manage the versions of:
- `flutter`
- `rust`

Add the plugins and install the versions needed using:
```bash
asdf plugin add flutter
asdf plugin add rust
asdf install flutter 3.22.3-stable # around 1.2GB
asdf install flutter 3.24.5-stable # around 1.4GB
asdf install rust 1.75.0
asdf install rust 1.81.0
```

#### Flutter & Rust Version Profiles

In this setup I use two tooling setups:
- `build1`: Used for building the bridge
- `build2`: Used for building the app

Create file `.tool-versions-build1` with this content:
```text
rust 1.75.0
flutter 3.22.3-stable
ruby system
```

And `.tool-versions-build2` with this content:
```text
rust 1.81.0
flutter 3.24.5-stable
ruby system
```

> [!TIP]
> You can also use the `cmake` and `nasm` plugins and use the following lines to manage their versions:
> ```text
> cmake 3.29.7
> nasm 2.14rc1
> ```

#### Switch Between `build1` & `build2`

Create a script `activate_tooling.sh` which can be passed either `1` or `2` as an argument:

```bash
ln -sF ".tool-versions-build$1" .tool-versions

if [ $1 == 1 ]; then
    pushd flutter 
    sed -i -e 's/extended_text: 14.0.0/extended_text: 13.0.0/g' pubspec.yaml
    flutter pub get
    popd
fi

if [ $1 == 2 ]; then
    pushd flutter 
    sed -i -e 's/extended_text: 13.0.0/extended_text: 14.0.0/g' pubspec.yaml
    flutter pub get
    popd
fi
```

Create another script `./show_tooling.sh` to show you which tooling setup is active:

```bash
printf "You are using the \e[33;1m%s\e[0m tooling.\n" $(basename $(readlink -f .tool-versions) | sed -e 's/\.tool-versions-//')
```

#### Patch Flutter 3.24.5-stable

```bash
git -C ~/.asdf/installs/flutter/3.24.5-stable apply ./.github/patches/flutter_3.24.4_dropdown_menu_enableFilter.diff
```

#### Build Bridge using `build1` Tooling

```bash
./activate_tooling.sh 1
cargo install flutter_rust_bridge_codegen --version 1.80.1 --features "uuid" --locked
asdf reshim rust
flutter_rust_bridge_codegen --rust-input ./src/flutter_ffi.rs --dart-output ./flutter/lib/generated_bridge.dart --c-output ./flutter/macos/Runner/bridge_generated.h
```

#### Build App using `build2` Tooling

```bash
./activate_tooling.sh 2
./build.py --flutter --screencapturekit
ln -sF $PWD/flutter/build/macos/Build/Products/Release/RustDesk.app RustDesk.app
```
